File: 1_two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    solution = Hash.new
    
    nums.each_with_index do |num, idx|
        if solution[num]
            return [solution[num], idx]
        else
            solution[target-num] = idx
        end
    end
end

File: 1_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let complements = {};

  for (let i = 0; i < nums.length; ++i) {
    if (complements[nums[i]] !== undefined) {
      return [complements[nums[i]], i];
    } else {
      complements[target - nums[i]] = i;
    }
  }

  return null;
};